src/java/org/apache/cassandra/db/tries/InMemoryTrie.java

```diff
@@ -129,12 +129,40 @@ enum ExpectedLifetime
         switch (lifetime)
         {
             case SHORT:
-                cellAllocator = new MemoryAllocationStrategy.NoReuseStrategy(this::allocateNewCell);
-                objectAllocator = new MemoryAllocationStrategy.NoReuseStrategy(this::allocateNewObject);
+                cellAllocator = new MemoryAllocationStrategy.NoReuseStrategy(new MemoryAllocationStrategy.Allocator()
+                {
+                    @Override
+                    public int allocate() throws TrieSpaceExhaustedException
+                    {
+                        return allocateNewCell();
+                    }
+                });
+                objectAllocator = new MemoryAllocationStrategy.NoReuseStrategy(new MemoryAllocationStrategy.Allocator()
+                {
+                    @Override
+                    public int allocate() throws TrieSpaceExhaustedException
+                    {
+                        return allocateNewObject();
+                    }
+                });
                 break;
             case LONG:
-                cellAllocator = new MemoryAllocationStrategy.OpOrderReuseStrategy(this::allocateNewCell, opOrder);
-                objectAllocator = new MemoryAllocationStrategy.OpOrderReuseStrategy(this::allocateNewObject, opOrder);
+                cellAllocator = new MemoryAllocationStrategy.OpOrderReuseStrategy(new MemoryAllocationStrategy.Allocator()
+                {
+                    @Override
+                    public int allocate() throws TrieSpaceExhaustedException
+                    {
+                        return allocateNewCell();
+                    }
+                }, opOrder);
+                objectAllocator = new MemoryAllocationStrategy.OpOrderReuseStrategy(new MemoryAllocationStrategy.Allocator()
+                {
+                    @Override
+                    public int allocate() throws TrieSpaceExhaustedException
+                    {
+                        return allocateNewObject();
```

Comment:
This one does not throw an exception, same for the other occurrence

Reply:
good catch. fixed